modules/sampler_hijack.py

```diff
@@ -257,15 +264,15 @@ def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor) -> to
                 # so the match is at least of length 1.
                 match_length = 1
 
-                # Extend the match backwards as far as possible.
-                while True:
+                # Extend the match backwards (at most to 1000 to prevent exponent overflow at penalty calculation) (this cap also improves performance on worst case)
+                while match_length < 1000:
```

Comment:
The cap can be **way** lower than 1000. Even 50 should be more than enough. The default `dry_base` is 1.75, and 1.75^50 > 10^12, which is a penalty that makes the token probability equal to zero for all practical purposes.

At which point I guess the Z algorithm becomes unnecessary, because the function is no longer quadratic even in the worst case, since the inner loop is bounded by a constant.

Reply:
Correct

Reply:
I'd say let's merge this PR and close the other one?

Reply:
Yes, that's the most pragmatic solution IMO. I'd file the Z algorithm under "it's cool that it's possible, but the juice isn't quite worth the squeeze in this case".